app/views/ShareView/index.js

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+	View, Text, TextInput, Image
+} from 'react-native';
+import { connect } from 'react-redux';
+import ShareExtension from 'rn-extensions-share';
+
+import {
+	COLOR_TEXT_DESCRIPTION
+} from '../../constants/colors';
+import I18n from '../../i18n';
+import RocketChat from '../../lib/rocketchat';
+import { CustomIcon } from '../../lib/Icons';
+import log from '../../utils/log';
+import styles from './styles';
+import Loading from './Loading';
+import database from '../../lib/realm';
+import { CustomHeaderButtons, Item } from '../../containers/HeaderButton';
+import { isReadOnly, isBlocked } from '../../utils/room';
+
+@connect(state => ({
+	username: state.login.user && state.login.user.username
+}))
+export default class ShareView extends React.Component {
+	static navigationOptions = ({ navigation }) => {
+		const canSend = navigation.getParam('canSend', false);
+
+		return ({
+			title: I18n.t('Share'),
+			headerRight:
+				canSend
+					? (
+						<CustomHeaderButtons>
+							<Item
+								title={I18n.t('Send')}
+								onPress={navigation.getParam('sendMessage')}
+								testID='send-message-share-view'
+								buttonStyle={styles.send}
+							/>
+						</CustomHeaderButtons>
+					)
+					: null
+		});
+	}
+
+	static propTypes = {
+		navigation: PropTypes.object,
+		username: PropTypes.string.isRequired
+	};
+
+	constructor(props) {
+		super(props);
+		const { navigation, username } = this.props;
+		const rid = navigation.getParam('rid', '');
+		const name = navigation.getParam('name', '');
+		const value = navigation.getParam('value', '');
+		const isMedia = navigation.getParam('isMedia', false);
+		const fileInfo = navigation.getParam('fileInfo', {});
+
+		this.rooms = database.objects('subscriptions').filtered('rid = $0', rid);
+
+		this.state = {
+			rid,
+			value,
+			isMedia,
+			name,
+			fileInfo,
+			loading: false,
+			room: this.rooms[0] || { rid },
+			user: { username },
+			file: {
+				name: fileInfo ? fileInfo.name : '',
+				description: ''
+			}
+		};
+	}
+
+	componentDidMount() {
+		const { room, user } = this.state;
+		const { navigation } = this.props;
+		navigation.setParams({ sendMessage: this._sendMessage, canSend: !(isReadOnly(room, user) || isBlocked(room)) });
+	}
+
+	bytesToSize = bits => `${ ((bits / 8) / 1048576).toFixed(2) }MB`;
+
+	_sendMessage = async() => {
+		const { isMedia } = this.state;
+		this.setState({ loading: true });
+
+		if (isMedia) {
+			await this.sendMediaMessage();
+		} else {
+			await this.sendTextMessage();
+		}
+
+		this.setState({ loading: false });
+		ShareExtension.close();
```

Comment:
Will this never fail or should we `try/catch` it?

Reply:
never fail.

Reply:
never fail